From 605c53ad0f209b9f62a15c3f89a9930860153489 Mon Sep 17 00:00:00 2001
From: Stefano Lattarini <stefano.lattarini@gmail.com>
Date: Thu, 30 Aug 2012 18:53:14 +0200
Subject: [PATCH 11/89] build: don't list files with unknown suffix in _SOURCES

While this works with mainline Automake (which blindly treats source
files with an unknown extension as if they were header files), it is
undocumented (albeit admittedly unlikely to change).  Moreover, it no
longer works with Automake-NG (and that's by design), so the use of
such feature would make a port to Automake-NG more difficult.

* libparted/labels/Makefile.am (liblabels_la_SOURCES): ... don't
list 'pt-limit.gperf' here; rather ...
(EXTRA_DIST): ... list 'pt-limit.c' and 'pt-limit.gperf' here.
(EXTRA_DIST, BUILT_SOURCES): Enhance few comments.

Copyright-paperwork-exempt: yes
---
 libparted/labels/Makefile.am | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/libparted/labels/Makefile.am b/libparted/labels/Makefile.am
index ec8abeb..f4e4d27 100644
--- a/libparted/labels/Makefile.am
+++ b/libparted/labels/Makefile.am
@@ -39,10 +39,11 @@ liblabels_la_LIBADD = $(OS_LIBS) $(INTLLIBS)
 
 AM_CPPFLAGS = $(partedincludedir) $(INTLINCS)
 
+# Included by 'pt-tools.c', so needs to be built early.
 BUILT_SOURCES = pt-limit.c
 MAINTAINERCLEANFILES = $(BUILT_SOURCES)
-liblabels_la_SOURCES += pt-limit.gperf
-EXTRA_DIST = $(BUILT_SOURCES)
+# DOn't add this to '_SOURCES', because it's not to be compiled!
+EXTRA_DIST= pt-limit.c
 
 GPERF = gperf
 GPERF_OPTIONS = \
@@ -55,3 +56,4 @@ pt-limit.c: pt-limit.gperf
 	  > $@-tmp
 	chmod a-w $@-tmp
 	mv $@-tmp $@
+EXTRA_DIST += pt-limit.gperf
-- 
1.8.5.3

